### DISCUSSION TITLE
How to create a custom Input component

### DISCUSSION THREAD

**Discussion by jmdamato** [react-hook-form/react-hook-form#4096] [Q&A] [answered]
I am trying to create a custom Input component that will include the `input` tag along with the validation error, but the value entered into the input gets removed on validation. Here is a link to my codepen. https://codesandbox.io/s/react-hook-form-v6-validationschema-forked-vvjc8?file=/src/index.js

I also tried passing the `ref` inside `<Input />`, but that still doesn't work. 

To Test: Type something in the First Name field then click the Last Name field.

**Comment by nbnsresta** [comment, marked as answer]
The issue is that you are creating new reference of `Input` everytime you render the parent component. In turn the whole component type is reset while validating. 
Do this.
https://codesandbox.io/s/react-hook-form-v6-validationschema-forked-q9ogr?file=/src/index.js

- Separate the components properly.
- Pass error as individual prop
- Forward ref on your custom input so that the changes can be tracked directly by the inner input component

**Reply by bluebill1049** [reply]
Thanks very much @nbnsresta for answering those questions.
